### PR TITLE
This re-brands the `aptos-sf-stream` into `aptos-fh-stream` (`fh` as a shortcut of Firehose)

### PR DESCRIPTION
**Pull request by maoueh**
### Description

The package name and all files related to this functionality has been renamed from `sf` to `fh`.

Also, the top-level config element has been renamed from `sf_stream` to `firehose_stream` to make easier for new user to discover the functionality as `sf` or `fh` as little meaning and is hard to search on Google while at least Firehose should return some results.

Changed also some reference from `StreamingFast` to `StreamingFast Firehose` in some comments as well as fixing the `Streaming Fast` company name to be `StreamingFast`.

### Test Plan
- Build `aptos-node`, follow instructions in `fh-stream/README.md` about how to run it
- Ensure it's streaming firehose log to stdout

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3008)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -91,7 +91,7 @@ members = [
     "secure/storage",
     "secure/storage/github",
     "secure/storage/vault",
-    "sf-stream",
+    "fh-stream",
```
Comment (bowenyang007) [inline Cargo.toml:94]: Lint failed. This needs to be alphabetically ordered I think.

**Review by bowenyang007 — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (maoueh) [reply Cargo.toml]: Fixed and push, let's see now :)

**Review by gregnazario — APPROVED** [review]
Please rebase and regenerate golden files etc.

Did you want to rename the golden files folder to fh_v1?

**Comment by maoueh** [comment]
Rebased and indeed also renamed golden files `sf_v1` to `fh_v1`. Don't think golden re-generation is required, tests passes here within `aptos-fh-stream` crate.

**Pull request merged**